### PR TITLE
Make prow update-config a postsubmit job

### PR DESCRIPTION
**Pull request by krzyzacy**

### CODE REVIEW

**Comment by krzyzacy** [comment]
retrigger from prow is my good friend..

/assign @spxtr

File: jobs/config.json [hunk 1]
```diff
@@ -2601,6 +2601,13 @@
   "maintenance-ci-janitor": {
     "scenario": "kubernetes_janitor"
   }, 
+  "maintenance-ci-prow-config-upload": {
+    "args": [
+      "cd test-infra/prow", 
+      "make update-config"
```
Comment (spxtr) [inline jobs/config.json]: Also `make update-plugins`.

Comment (krzyzacy) [inline jobs/config.json]: done

File: prow/config.yaml [hunk 2]
```diff
@@ -710,6 +710,27 @@ postsubmits:
         secret:
           secretName: service-account
 
+  - name: maintenance-ci-prow-config-upload
+    branches:
+    - master
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e-prow:v20170418-c08e1094
```
Comment (spxtr) [inline prow/config.yaml]: Why use kubekins-e2e-prow for this? It's not running e2e tests.

File: prow/config.yaml [hunk 3]
```diff
@@ -710,6 +710,27 @@ postsubmits:
...
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /etc/service-account/service-account.json
+        volumeMounts:
+        - name: service
+          mountPath: /etc/service-account
+          readOnly: true
+      volumes:
+      - name: service
+        secret:
+          secretName: service-account
```
Comment (spxtr) [inline prow/config.yaml]: This service account won't have access to the prow cluster so it won't be able to update the cluster. That's a good thing. We'll need to add another account or do this in another way.

Comment (krzyzacy) [inline prow/config.yaml]: shall I bind the k8s-prow service account here?

File: prow/config.yaml (hunk 2 above)
Comment (krzyzacy) [inline prow/config.yaml]: good catch! done

**Comment by spxtr** [comment]
Does it make sense to use a prow plugin for this?

**Comment by krzyzacy** [comment]
> Does it make sense to use a prow plugin for this?

Hummm, what will be the pattern? Something looks for test-infra merges, then checkout head and run make commands?

It feels more natural as a postsubmit job, since we already do it for testgrid config.

**Comment by spxtr** [comment]
We wouldn't even checkout the repo, just grab the file and update the configmap using the kubernetes client that already exists in the plugin client.

**Comment by krzyzacy** [comment]
a plugin that scans for merge event and grabs a file and update prow humm.. sounds clear with a prow job.. Any concern with adding a different service account, like k8s-prow?

**Comment by spxtr** [comment]
It's not a big deal, just more complicated than it needs to be. Plugins already have authenticated access to the apiserver, no need for a service account.

**Comment by krzyzacy** [comment]
I'll try to write a plugin later

**Pull request closed without merging**